corehq/apps/accounting/invoicing.py

```diff
@@ -347,12 +347,13 @@ def __init__(self, date_start, date_end, account, recipients=None):
         self.subscriptions = {}
 
     def create_invoice(self):
-        for subscription in self.account.subscription_set.all():
-            if should_create_invoice(subscription, subscription.subscriber.domain, self.date_start, self.date_end):
-                if subscription.plan_version in self.subscriptions:
-                    self.subscriptions[subscription.plan_version].append(subscription)
+        for sub in self.account.subscription_set.all():
+            if not sub.do_not_invoice \
+                    and should_create_invoice(sub, sub.subscriber.domain, self.date_start, self.date_end):
+                if sub.plan_version in self.subscriptions:
```

Comment:
Totally optional, non-blocking nit, and I know you didn't write this ... but if line 347 was
```python
    self.subscriptions = defaultdict(list)
```
then lines 353 to 356 could be replaced with 
```python
    self.subscriptions[sub.plan_version].append(sub)
```